pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -98,75 +99,74 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 
 	var created bool
 	var user *userv1.User
-	if user, created, err = r.ensureUser(userAcc); err != nil {
-		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
+	if user, created, err = r.ensureUser(reqLogger, userAcc); err != nil || created {
 		return reconcile.Result{}, err
-	} else if created {
-		return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, "")
 	}
 
 	var identity *userv1.Identity
-	if identity, created, err = r.ensureIdentity(userAcc); err != nil {
-		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
+	if identity, created, err = r.ensureIdentity(reqLogger, userAcc); err != nil || created {
 		return reconcile.Result{}, err
-	} else if created {
-		return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, "")
 	}
 
-	if created, err = r.ensureMapping(userAcc, user, identity); err != nil {
-		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
+	if created, err = r.ensureMapping(reqLogger, userAcc, user, identity); err != nil || created {
 		return reconcile.Result{}, err
-	} else if created {
-		return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, "")
 	}
 
 	return reconcile.Result{}, r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioned, "")
 }
 
-func (r *ReconcileUserAccount) ensureUser(userAcc *toolchainv1alpha1.UserAccount) (*userv1.User, bool, error) {
+func (r *ReconcileUserAccount) ensureUser(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (*userv1.User, bool, error) {
 	user := &userv1.User{}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user); err != nil {
 		if errors.IsNotFound(err) {
-			log.Info("creating a new user", "name", userAcc.Name)
-			user = newUser(userAcc)
-			if err := controllerutil.SetControllerReference(userAcc, user, r.scheme); err != nil {
+			logger.Info("creating a new user", "name", userAcc.Name)
+			if err := r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error()); err != nil {
```

Comment:
do we really need to set the error here? I think that it is obvious that the `User` does/did not exist when the status is `provisioning`. I would use `error` field only when something fails.

Reply:
ah, I just figured out that the Nurali's [comment](https://github.com/codeready-toolchain/member-operator/pull/25/files#r298037991) is about the same :-)

Reply:
Addressed in https://github.com/codeready-toolchain/member-operator/pull/25/commits/81f9d57c0d73f11d0430e1ae57b28891038d1a96